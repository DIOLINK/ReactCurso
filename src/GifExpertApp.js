import React, { useState } from 'react';
import AddCategory from './components/AddCategory';
import GifGrid from './components/GifGrid';

const GifExpertApp = ({ categoriesDefault = [] }) => {
  const [categories, setCategories] = useState(categoriesDefault);

  return (
    <>
      <h2>GifExpertApp</h2>
      <AddCategory setCategories={setCategories} />
      <hr />

      <ol>
        {categories.map((category) => (
          <GifGrid key={category} category={category} />
        ))}
      </ol>
    </>
  );
};

export default GifExpertApp;
