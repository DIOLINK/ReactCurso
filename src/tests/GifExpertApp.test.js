import { shallow } from 'enzyme';
import GifExpertApp from '../GifExpertApp';

describe('Prueba de GifExpertApp', () => {
  const wrapper = shallow(<GifExpertApp />);
  it('Debe de mostrar el componente correctamente', () => {
    expect(wrapper).toMatchSnapshot();
  });
  it('Debe de mostrar una lista de categorias', () => {
    const categories = ['One Punch', 'Samurai X'];
    const wrapper = shallow(<GifExpertApp categoriesDefault={categories} />);
    expect(wrapper.find('GifGrid').length).toBe(categories.length);
  });
});
