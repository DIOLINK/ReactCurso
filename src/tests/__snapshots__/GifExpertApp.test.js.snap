// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba de GifExpertApp Debe de mostrar el componente correctamente 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
